iml-services/iml-device/src/main.rs

```diff
@@ -45,6 +50,140 @@ async fn create_cache(pool: &DbPool) -> Result<Cache, ImlDeviceError> {
     Ok(Arc::new(Mutex::new(data)))
 }
 
+async fn update_devices(pool: &DbPool, host: &Fqdn, devices: &Device) {
+    let device_to_insert = NewChromaCoreDevice {
+        fqdn: host.to_string(),
+        devices: serde_json::to_value(devices)
+            .expect("Could not convert incoming Devices to JSON."),
+    };
+
+    let new_device = diesel::insert_into(dvc::table)
+        .values(device_to_insert)
+        .on_conflict(dvc::fqdn)
+        .do_update()
+        .set(dvc::devices.eq(excluded(dvc::devices)))
+        .get_result_async::<ChromaCoreDevice>(&pool)
+        .await;
+    match new_device {
+        Ok(d) => {
+            tracing::info!("Inserted devices from host '{}'", host);
+            tracing::debug!("Inserted {:?}", d);
+        }
+        Err(e) => {
+            tracing::error!("Error saving new devices: {}", e);
+        }
+    }
+}
+
+async fn update_client_mounts(
+    pool: &DbPool,
+    ct_id: Option<i32>,
+    host: &Fqdn,
+    mounts: &HashSet<Mount>,
+) {
+    let host_id = match host::table
+        .select(host::id)
+        .filter(host::fqdn.eq(host.to_string()))
+        .get_result_async::<i32>(&pool)
+        .await
+    {
+        Ok(id) => {
+            tracing::debug!("Host id: {}", id);
+            id
+        }
+        Err(e) => {
+            tracing::error!("Host '{}' is unknown: {}", host, e);
```

Comment:
An error here doesn't just mean a host is unknown. 

It means a host is unknown or there was a db error. This is why `optional()` should be used.